File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int maxcnt = 0;
        int cnt = 0;
        for(int i = 0; i < n; i++) 
        {
            if(nums[i] == 1) 
            {
                cnt++;
            }
            else 
            {
                cnt = 0;
            }
            maxcnt = max(cnt, maxcnt);
        }
        return maxcnt;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int n = nums.length;
        int cnt = 0;
        int maxCnt = 0;
        for(int i = 0; i < n; i++)
        {
            if(nums[i] == 1)
            {
                cnt++;
            }
            else {
                cnt = 0;
            }
            if(cnt > maxCnt)
            {
                maxCnt = cnt;
            }
        }
        return maxCnt;
    }
}